src/internal/m365/collection/site/backup.go

```diff
@@ -185,17 +188,35 @@ func CollectLists(
 			el.AddRecoverable(ctx, clues.WrapWC(ctx, err, "creating list collection path"))
 		}
 
-		collection := NewPrefetchCollection(
+		lazyFetchCol := NewLazyFetchCollection(
 			bh,
 			dir,
-			ac,
-			scope,
 			su,
-			bpc.Options)
+			cl)
 
-		collection.AddItem(
+		lazyFetchCol.AddItem(
 			ptr.Val(list.GetId()),
-			ptr.Val(lmt))
+			modTime)
+
+		collection = lazyFetchCol
+
+		// Always use lazyFetchCol.
+		// In case we receive zero mod time from graph fallback to prefetchCol.
+		if modTime.IsZero() {
+			prefetchCol := NewPrefetchCollection(
+				bh,
+				dir,
+				ac,
+				scope,
+				su,
+				bpc.Options)
+
+			prefetchCol.AddItem(
```

Comment:
nit: `AddItem` call can be moved out as common code for both lazy and prefetch col.